src/main/java/FrameworksDrivers/ScheduleDrawing.java

```diff
@@ -0,0 +1,179 @@
+                    xcord = 125 + (i * 100);
+                }
+            }
+
+            int yStartCord = 0;
+            for (int i = 0; i < 24; i++){
+                if (Objects.equals(schedule.get(j).getEventStartTime(), timesList[i])){
+                    yStartCord = 140 + (i * 25);
+                }
+            }
+
+            int yEndCord = 0;
+            for (int i = 0; i < 24; i++){
+                if (Objects.equals(schedule.get(j).getEventEndTime(), timesList[i])){
+                    yEndCord = 140 + (i * 25);
+                }
+            }
+            System.out.println(xcord);
+            System.out.println(yStartCord);
+            System.out.println(yEndCord);
+
+            JButton eventbutton = new JButton(schedule.get(j).getEventName());
+
+            eventbutton.setBounds(xcord,yStartCord,100,(yEndCord-yStartCord));
+            eventbutton.addActionListener(editEventListener);
+            add(eventbutton);
+            this.buttons.put(eventbutton, schedule.get(j));
+//            g.setColor(Color.PINK); // different color for each day slot
+//            g.fillRect(xcord, yStartCord, 100, (yEndCord - yStartCord));
+//            g.setColor(Color.BLACK);
+//            g.drawString(schedule.get(j).getEventName(), (xcord + 10), (yStartCord + 15));
+        }
+    }
+
+    private class EditEventListener implements ActionListener {
+        @Override
+        public void actionPerformed(ActionEvent e) {
+
+            Object source = e.getSource();
+
+            for (JButton button: buttons.keySet()) {
+                if (source == button) {
+
+                    Object[] addEventText = {"Name:", addEventName, "Date:",
+                            addEventDate, "Start time:", addEventStart, "End time:", addEventEnd};
+
+                    int buttonChoice = JOptionPane.showConfirmDialog(null,
+                            addEventText, "Edit Entities.Event", JOptionPane.OK_CANCEL_OPTION);
+
+                    if (buttonChoice == JOptionPane.OK_OPTION) {
+                        // will interact with Entities.Event class
+
+                        // temporarily assigning the values to temporary variables
+                        String eventName = addEventName.getText();
+                        String eventDate = (String) addEventDate.getSelectedItem();
+
+                        String eventStart = (String) addEventStart.getSelectedItem();
+                        String eventStartString = eventStart.split(":")[0];
+                        int eventStartInt = Integer.parseInt(eventStartString);
+
+                        String eventEnd = (String) addEventEnd.getSelectedItem();
+                        String eventEndString = eventEnd.split(":")[0];
+                        int eventEndInt = Integer.parseInt(eventEndString);
+
+
+                        Event event = new Event(eventName, eventDate, eventStartInt, eventEndInt);
```

Comment:
we're not making events this way anymore either, right? (ignore this if this is purely for testing purposes)